subprojects/core/src/main/java/org/gradle/api/internal/component/SoftwareComponentInternal.java

```diff
@@ -24,5 +24,15 @@
  * This will be replaced by {@link org.gradle.api.component.ComponentWithVariants} and other public APIs.
  */
 public interface SoftwareComponentInternal extends SoftwareComponent {
+    /**
+     * This should be called {@code getVariants} and return a set of {@link org.gradle.api.component.SoftwareComponentVariant}s,
+     * though this is currently used by KMP and perhaps AGP due to there being a missing public API. However, renaming is
+     * difficult since {@link org.gradle.api.component.ComponentWithVariants} already defines a {@code getVariants} method.
+     * The existing {@code getVariants} method however is misnamed and should probably be called {@code getChildren}, and its
+     * interface called {@code ComponentWithChildren}. The existing {@code ComponentWithVariants} actually defines an API for
+     * declaring child components, which contribute their variants as defined by this method, as remote variants in GMM.
+     * <p>
+     * This method and {@link UsageContext} should both be deprecated in favor of a new public API.
+     */
     Set<? extends UsageContext> getUsages();
```

Comment:
We can't define a `getVariants` method here, since `ComponentWithVariants` also defines a `getVariants` method with a different return type -- and there are multiple first and third party `SoftwareComponent` implementations which implement both this interface and `ComponentWithVariants`.

Reply:
I forgot about that last part.  We could name it `getAllVariants()` or something slightly different then, it'll still make the usage clearer.

Reply:
Yeah that was one of my original thoughts. But I would rather not compromise on this method name. Its going to be important in the DSL. So if we're eventually going to find yet another name for it, I would rather not push the existing implementers to change to the new method name if its eventually just going to change _again_.